framework/audio/iaudioengine.h

```diff
@@ -25,53 +25,29 @@
 
 #include "ret.h"
 #include "async/channel.h"
-#include "iaudiosource.h"
-#include "audiotypes.h"
+#include "isequencer.h"
+#include "midi/isynthesizer.h"
 
 namespace mu {
 namespace audio {
+class Mixer;
+class IAudioDriver;
+
 class IAudioEngine : MODULE_EXPORT_INTERFACE
 {
     INTERFACE_ID(IAudioEngine)
 
 public:
     virtual ~IAudioEngine() = default;
 
-    using handle = unsigned int;
-    using time = float;
-
-    enum class Status {
-        Undefined = 0,
-        Created,
-        Playing,
-        Paused,
-        Stoped
-    };
-
-    virtual Ret init() = 0;
+    virtual Ret init(std::shared_ptr<IAudioDriver>) = 0;
     virtual void deinit() = 0;
     virtual bool isInited() const = 0;
     virtual async::Channel<bool> initChanged() const = 0;
-
-    virtual float sampleRate() const = 0;
-
-    virtual handle play(std::shared_ptr<IAudioSource> src, float volume = -1, float pan = 0, bool paused = false) = 0;
-    virtual void seek(handle h, time sec) = 0;
-    virtual void setPause(handle h, bool paused) = 0;
-    virtual void stop(handle h) = 0;
-
-    virtual Status status(handle h) const = 0;
-    virtual async::Channel<Status> statusChanged(handle h) const = 0;
-
-    virtual time position(handle h) const = 0;
-
-    virtual void setVolume(handle h, float volume) = 0; // 0. - 1.
-    virtual void setPan(handle h, float val) = 0; // -1 only left, 0 center, 1 only right
-    virtual void setPlaySpeed(handle h, float speed) = 0;
-
-    // actions on driver callback
-    virtual void swapPlayContext(handle h, Context& ctx) = 0;
-    virtual async::Channel<Context> playContextChanged(handle h) const = 0;
+    virtual unsigned int sampleRate() const = 0;
+    virtual std::shared_ptr<ISequencer> sequencer() const = 0;
+    virtual unsigned int startSynthesizer(std::shared_ptr<midi::ISynthesizer> s) = 0;
```

Comment:
Why does this method return int?

Reply:
It doesn't, it returns an `unsigned int`:-)

Reply:
@RomanPudashkin it is a channel number in the mixer

Reply:
Roman means that it is better to return true/false, because it is not clear what kind of number it will return. You can use Ret. If you need to return a value then RetVal<int>. But as a user, it is difficult for me to understand what kind of int it is. Better to make alias ChannelNumber, then there will be something like RetVal<ChannelNumber>.

Reply:
fixed, added IMixer::ChannelID type https://github.com/musescore/MuseScore/pull/7274